### DISCUSSION TITLE
incompatible types: text[] and text?

### DISCUSSION THREAD

**Discussion by SupertigerDev** [prisma/prisma#10862] [Q&A]
![image](https://user-images.githubusercontent.com/8193841/147372952-f69cfd3a-7914-474a-8a3a-f906bcb30847.png)
I am getting this error when trying to do this:
```
model User {
  id String @id

  recipientChannels Channel[]

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}


model Channel {
  id String  @id

  recipientIds String[]
  recipients User[] @relation(fields: [recipientIds], references: [id])

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```
any help? thanks

**Comment by colinpcurtis** [comment]
I think the error is that `recipients` is a relation to the `User` table, so the reference in the `Channel` table should be of the same type as the `@id` field in the `User` channel, in your case a `String`.

Thus, try changing your `Channel` model to

```prisma
model Channel {
  id String  @id

  recipientIds String
  recipients User[] @relation(fields: [recipientIds], references: [id])

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```

**Reply by SupertigerDev** [reply]
but i want recipientIds to be an array of strings hmm.

**Reply by colinpcurtis** [reply]
I would recommend looking at the data in prisma studio.  You'll see that you don't actually need to have the array of strings to store the relation, and when you do a find query you'll still see all the recipients when you get the data.

**Reply by SupertigerDev** [reply]
So can I do prisma.channels.create({recipientIds: [id1, id2]}) and it will work?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
